functions/[[index]].ts

```diff
@@ -11,7 +19,10 @@ export const onRequest: PagesFunction = async ({ request, next }) => {
   }
   const res = next()
   try {
-    return new HTMLRewriter().on('head', new MetaTagInjector(data, request)).transform(await res)
+    const content = new HTMLRewriter().on('head', new MetaTagInjector(data, request)).transform(await res).body
+    return new Response(content, {
+      status: doesMatchPath(requestURL.pathname) || requestURL.pathname.includes('.') ? 200 : 404,
```

Comment:
what cases does `requestURL.pathname.includes('.')` cover?

Reply:
this is intended to allow filenames. after the initial request returns `index.html` react starts fetching all the other assets that the page needs. afact, they all have file extensions